src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -151,5 +221,88 @@ public Task<TagHelperContent> GetChildContentAsync(bool useCachedResult)
         {
             return _getChildContentAsync(useCachedResult);
         }
+
+        /// <inheritdoc />
+        public void WriteTo(TextWriter writer, HtmlEncoder encoder)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            _preElement?.WriteTo(writer, encoder);
+
+            var isTagNameNullOrWhitespace = string.IsNullOrWhiteSpace(TagName);
+
+            if (!isTagNameNullOrWhitespace)
+            {
+                writer.Write('<');
+                writer.Write(TagName);
+
+                foreach (var attribute in Attributes)
+                {
+                    writer.Write(' ');
+                    writer.Write(attribute.Name);
+
+                    if (attribute.Minimized)
+                    {
+                        continue;
+                    }
+
+                    writer.Write("=\"");
+                    var value = attribute.Value;
+                    var htmlContent = value as IHtmlContent;
+                    if (htmlContent != null)
+                    {
+                        // There's no way of tracking the attribute value quotations in the Razor source. Therefore, we
+                        // must escape any IHtmlContent double quote values in the case that a user wrote:
+                        // <p name='A " is valid in single quotes'></p>
+                        //
+                        // This is really not optimal for perf, but it's the best we can do for right now.
+                        using (var stringWriter = new StringWriter())
+                        {
+                            htmlContent.WriteTo(stringWriter, encoder);
+
+                            var stringValue = stringWriter.ToString();
+                            stringValue = stringValue.Replace("\"", "&quot;");
+
+                            writer.Write(stringValue);
+                        }
+                    }
+                    else if (value != null)
+                    {
+                        var encodedValue = encoder.Encode(value.ToString());
+                        writer.Write(encodedValue);
+                    }
+
+                    writer.Write('"');
+                }
+
+                if (TagMode == TagMode.SelfClosing)
+                {
+                    writer.Write(" /");
+                }
+
+                writer.Write('>');
+            }
+
+            if (isTagNameNullOrWhitespace || TagMode == TagMode.StartTagAndEndTag)
+            {
+                _preContent?.WriteTo(writer, encoder);
+
+                _content?.WriteTo(writer, encoder);
```

Comment:
Nope, because if you suppress the content (`.Content.Clear()`) and then go ahead and append to it; any appended content will override the suppressed content and will result in content being rendered.